lib/tests/test_git.rs

```diff
@@ -2330,6 +2330,68 @@ fn test_fetch_initial_commit() {
     );
 }
 
+#[test]
+fn test_fetch_initial_commit_head_is_set() {
+    let test_data = GitRepoData::create();
+    let git_settings = GitSettings {
+        auto_local_bookmark: true,
+        ..Default::default()
+    };
+    let initial_git_commit = empty_git_commit(&test_data.origin_repo, "refs/heads/main", &[]);
+    test_data.origin_repo.set_head("refs/heads/main").unwrap();
+    let new_git_commit = empty_git_commit(
+        &test_data.origin_repo,
+        "refs/heads/main",
+        &[&initial_git_commit],
+    );
+    test_data
+        .origin_repo
+        .reference("refs/tags/v1.0", new_git_commit.id(), false, "")
+        .unwrap();
+
+    let mut tx = test_data.repo.start_transaction(&test_data.settings);
+    let stats = git::fetch_and_get_default_branch(
+        tx.repo_mut(),
+        &test_data.git_repo,
+        "origin",
+        &[StringPattern::everything()],
+        git::RemoteCallbacks::default(),
+        &git_settings,
+        None,
+    )
+    .unwrap();
+    assert_eq!(stats.default_branch, Some("main".to_string()));
+    assert!(stats.import_stats.abandoned_commits.is_empty());
+    let repo = tx.commit("test").unwrap();
+    // The new commit visible after git::fetch_and_get_default_branch().
+    let view = repo.view();
+    assert!(view.heads().contains(&jj_id(&new_git_commit)));
+    let commit_target = RefTarget::normal(jj_id(&new_git_commit));
+    let commit_remote_ref = RemoteRef {
+        target: commit_target.clone(),
+        state: RemoteRefState::Tracking,
+    };
+    assert_eq!(
+        *view.git_refs(),
+        btreemap! {
+            "refs/remotes/origin/main".to_string() => commit_target.clone(),
+            "refs/tags/v1.0".to_string() => commit_target.clone(),
```

Comment:
nit: maybe we can omit uninteresting assertions and refs? I think here we just need to test `default_branch` and do some basic sanity checks.

Reply:
Makes sense, since they're already verified in the test above. I gone ahead to whack the asserts after verifying `default_branch`.